cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -311,6 +315,16 @@ func (s *Server) Run(queue *Queue) {
 		}
 	}()
 
+	go func() {
+		for range jitterTicker(s.Vacuum, syscall.SIGUSR1) {
+			if zoekt.TombstonesEnabled(s.IndexDir) {
+				muIndexDir.Lock()
```

Comment:
I assume vacuum can be quite slow? Seems bad to be holding the lock while it is running?

Reply:
I think it will be slow but we also don't run it very often. I could imagine that runnig it once per day could be enough so I think holding the lock might be fine?

Reply:
while we hold it nothing else runs though right? IE your indexes will be stale. It probably is acceptable, but I wonder if we should just grab the lock each time we call vacuum on a compound shard instead so we atleast don't block the whole time.

Reply:
ok, I updated the locking logic. Now we lock each time we call `removeTombstones` on a compound shard. I had to move the mutex to the server struct which feels a bit odd, but it beats having a pointer to the mutex as a parameter of `vacuum`.